plugins/linux-sleep/fu-plugin-linux-sleep.c

```diff
@@ -1,43 +0,0 @@
-/*
- * Copyright (C) 2020 Richard Hughes <richard@hughsie.com>
- *
- * SPDX-License-Identifier: LGPL-2.1+
- */
-
-#include "config.h"
-
-#include "fu-plugin-vfuncs.h"
-#include "fu-hash.h"
-
-void
-fu_plugin_init (FuPlugin *plugin)
-{
-	fu_plugin_set_build_hash (plugin, FU_BUILD_HASH);
-	fu_plugin_add_rule (plugin, FU_PLUGIN_RULE_REQUIRED_FOR, "hsi");
-}
-
-gboolean
-fu_plugin_add_security_attrs (FuPlugin *plugin, GPtrArray *attrs, GError **error)
-{
-	FwupdSecurityAttr *attr;
-	gsize bufsz = 0;
-	g_autofree gchar *buf = NULL;
-	g_autoptr(GFile) file = g_file_new_for_path ("/sys/power/mem_sleep");
-
-	/* load file */
-	if (!g_file_load_contents (file, NULL, &buf, &bufsz, NULL, error)) {
-		g_autofree gchar *fn = g_file_get_path (file);
-		g_prefix_error (error, "could not open %s: ", fn);
-		return FALSE;
-	}
-	attr = fwupd_security_attr_new ("org.kernel.CheckS3Sleep");
-	fwupd_security_attr_set_level (attr, FWUPD_SECURITY_ATTR_LEVEL_THEORETICAL);
-	fwupd_security_attr_set_name (attr, "Linux Kernel S3 Sleep");
-	if (g_strstr_len (buf, bufsz, "deep") == NULL) {
-		fwupd_security_attr_add_flag (attr, FWUPD_SECURITY_ATTR_FLAG_SUCCESS);
-	} else {
-		fwupd_security_attr_set_result (attr, "Deep sleep available");
-	}
```

Comment:
So I don't think this one should go "away". I think we should have it for runtime checks - specifically if the firmware was set to default S2I, but the user changed it to `deep` via `/sys/power/mem_sleep` then I think that should fail a runtime check and you no longer trust things.

Reply:
Well two ideas.

1. Move the check into the facp plugin rather than a dedicated sleep plugin.  Create the s3 attribute only if facp set appropriately.
2. Keep sleep it's own plugin, and make anything that is set to s3 a runtime error on all systems set this way.

Reply:
Hang on, can't we just say to get HSI:3 we have to have `/sys/power/mem_sleep != [deep]` AND facp with LOW_POWER_S0_IDLE_CAPABLE set?

Reply:
Certainly - but the first one of those is a runtime thing that you can potentially change.

Reply:
> that you can potentially change

Right, but as the user can with secure boot too... I think we want to limit the number of things that trigger `!` especially when the admin setting [deep] manually is probably deliberate.

Reply:
I know of a few machines that have shipped out of the factory with a kernel command line option setting it to `deep`.  It's not because `s2idle` doesn't work, but specifically because of energy consumption requirements in a local market (IE something like component XYZ uses XXX mW on Windows and uses YYY mW on Linux, but that keeps SOC from hitting this state which makes whole platform use ZZZ W which means it doesn't meet local energy regulations).  This is firmly in the camp of driver is missing a feature or has a bug.

I think for most people that's a net positive tradeoff, but it should be viewed negatively by the `security` command.  However then again those things were set as shipped, not modified by the user.

TL;DR You're right.  Check it once at fwupd startup and that's enough.

Reply:
Okay, sleep plugin re-instated